lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php

```diff
@@ -666,20 +661,32 @@ function( $pseudo_selector ) use ( $selector ) {
 			}
 		}
 
-		/*
-		 * Reset default browser margin on the root body element.
-		 * This is set on the root selector **before** generating the ruleset
-		 * from the `theme.json`. This is to ensure that if the `theme.json` declares
-		 * `margin` in its `spacing` declaration for the `body` element then these
-		 * user-generated values take precedence in the CSS cascade.
-		 * @link https://github.com/WordPress/gutenberg/issues/36147.
-		 */
+		$block_rules = '';
+
+		// 2. Generate and append the rules that use the general selector.
 		if ( static::ROOT_BLOCK_SELECTOR === $selector ) {
+			/*
+			 * Reset default browser margin on the root body element.
+			 * This is set on the root selector **before** generating the ruleset
+			 * from the `theme.json`. This is to ensure that if the `theme.json` declares
+			 * `margin` in its `spacing` declaration for the `body` element then these
+			 * user-generated values take precedence in the CSS cascade.
+			 * @link https://github.com/WordPress/gutenberg/issues/36147.
+			 */
 			$block_rules .= 'body { margin: 0; }';
-		}
 
-		// 2. Generate and append the rules that use the general selector.
-		$block_rules .= static::to_ruleset( $selector, $declarations );
+			/*
+				Style engine is generating the root styles only.
+				This way we can iteratively introduce it to start generating styles.elements, and styles.blocks.
+			*/
+			$block_rules .= gutenberg_style_engine_generate_global_styles(
+				$node,
+				array( 'selector' => $selector )
+			);
+
+		} else {
```

Comment:
I like the simple if/else to figure out how we can incrementally switch over to the style engine by just tackling root to begin with 👍